tools/toolchains/arm.py

```diff
@@ -312,6 +313,11 @@ def __init__(self, target, *args, **kwargs):
         if target.core == "Cortex-M23" or target.core == "Cortex-M33":
             self.flags['common'].append("-mcmse")
 
+        # Create Secure library
+        if target.core == "Cortex-M23" or self.target.core == "Cortex-M33":
+            secure_file = join(build_dir, "cmse_lib.o")
```

Comment:
why is this in the tools and not behind a TARGET_M23 folder?

Reply:
Creation of secure library shouldn;t be part of tools?

Reply:
I dont know the workflow or modifications needed to that file

Reply:
@deepikabhavnani As I read this ld flag: http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0803h/pge1444636329852.html - this is the output thus located into build dir as cmse_lib.o. Thus it;s part of tools, or is there any other option?

cc @theotherjimmy

Reply:
@0xc0170 - Generation of Secure library has to per part of tools, ARMC6 compiler allows creation from elf file as well, but GCC/ARM support generation of this object file during linking.

We can rename lib to anything we want (like target specific) and can place in any required build folder (Current: .\BUILD\<Target Device>\<Toolchain>\cmse_lib.o